vertical-pod-autoscaler/pkg/recommender-external/input/metrics/external_metrics_client.go

```diff
@@ -49,23 +49,22 @@ type metricsClient struct {
 
 // GetExternalMetric gets all the values of a given external metric
 // that match the specified selector.
-// TODO: Review all that when we actually use it
-func (c *metricsClient) GetExternalMetric(metricName, namespace string, selector labels.Selector) ([]int64, time.Time, error) {
+func (c *metricsClient) GetExternalMetric(metricName, namespace string, selector labels.Selector) (int64, time.Time, error) {
 	metrics, err := c.client.NamespacedMetrics(namespace).List(metricName, selector)
 	recommender_metrics.RecordMetricsServerResponse(err, c.clientName)
 	if err != nil {
-		return []int64{}, time.Time{}, fmt.Errorf("unable to fetch metrics from external metrics API: %v", err)
+		return 0, time.Time{}, fmt.Errorf("unable to fetch metrics from external metrics API: %v", err)
 	}
 
 	if len(metrics.Items) == 0 {
-		return nil, time.Time{}, fmt.Errorf("no metrics returned from external metrics API")
+		return 0, time.Time{}, fmt.Errorf("no metrics returned from external metrics API")
 	}
 
-	res := make([]int64, 0)
-	for _, m := range metrics.Items {
-		res = append(res, m.Value.MilliValue())
-	}
-	timestamp := metrics.Items[0].Timestamp.Time
-	// TODO: Change (or wrap) this function to return Resources instead.
-	return res, timestamp, nil
+	klog.V(6).Infof("Got %d points: %+v", len(metrics.Items), metrics.Items)
```

Comment:
To avoid log flooding one day in case it returns tons of values:
`klog.V(6).Infof("Got %d points, keeping only the last one %+v", len(metrics.Items), metrics.Items[idx])`

Shouldn't we log a warning (or error) in case we have more than 1 point ?

Reply:
ok doing this